composer/algorithms/gated_linear_units/gated_linear_units.py

```diff
@@ -0,0 +1,159 @@
+    assert isinstance(
+        layer, BertOutput
+    ), 'The replacement policy will look for all instances of transformers.models.bert.modeling_bert.BertOutput'
+    return BERTGatedOutput(d_embed=layer.dense.out_features,
+                           d_ff=layer.dense.in_features,
+                           dropout_rate=layer.dropout.p,
+                           act_fn=act_fn,
+                           layernorm_eps=layer.LayerNorm.eps,
+                           wi_0_bias=wi_0_bias,
+                           wi_1_bias=wi_1_bias)
+
+
+def from_BertIntermediate(layer: torch.nn.Module, module_index: int) -> DummyBERTIntermediateOutput:
+    """Defines a replacement policy from a `transformers.models.bert.modeling_bert.BertIntermediate` to a `composer.algorithms.gated_linear_units.gated_linear_unit_layers.DummyBERTIntermediateOutput`"""
+    assert isinstance(
+        layer, BertIntermediate
+    ), 'The replacement policy will look for all instances of transformers.models.bert.modeling_bert.BertIntermediate'
+    return DummyBERTIntermediateOutput()
+
+
+def apply_gated_linear_units(model: torch.nn.Module,
+                             optimizers: Union[torch.optim.Optimizer, Sequence[torch.optim.Optimizer]],
+                             act_fn: Optional[Callable] = None,
+                             wi_0_bias: bool = False,
+                             wi_1_bias: bool = False) -> None:
+    """
+    Replaces the Linear layers in the feed-forward network with `Gated Linear Units <https://nvidia.github.io/apex/layernorm.html>`_.
+
+    Args:
+        model (:class:`torch.nn.Module`): the model to modify in-place
+        optimizers (:class:`torch.optim.Optimizer` | Sequence[:class:`torch.optim.Optimizer`], optional):
+            Existing optimizers bound to ``model.parameters()``. All optimizers that have already been
+            constructed with ``model.parameters()`` must be specified here so that
+            they will optimize the correct parameters.
+
+            If the optimizer(s) are constructed *after* calling this function,
+            then it is safe to omit this parameter. These optimizers will see the correct
+            model parameters.
+        act_fn (Callable, optional): Optionally, the activation function to use. If ``None``, the algorithm will
+            use the existing activation function in the model.
+        wi_0_bias (bool, optional): Whether to use biases in the linear layers within the GLU. Default: ``False``.
+        wi_1_bias (bool, optional): Whether to use biases in the linear layers within the GLU. Default: ``False``.
+    """
+    check_if_transformers_installed(TRANSFORMERS_INSTALLED)
+
+    # ensure that the model is an instance of a BERTModel, since our replacement policy is only defined for BERTs
+    if not isinstance(model, BERTModel):
```

Comment:
It would -- but we need to make sure that their BertIntermediate and BertOutput have the right overall structure (ie. it could be dangerous if their ATTN network does an operation that we consider part of our `BertGatedOutput`. 

For instance, HF had defined this class called [BertIntermediate](https://github.com/huggingface/transformers/blob/main/src/transformers/models/bert/modeling_bert.py#L437) that has what we would consider today as part of the FF network. 

This heavy-handed approach is to make sure that we can provide some sanity about the surgery-ified model, and that the composite model doesn't have characteristics that don't make sense for training. Part of that is knowing what their ATTN network looks like, so that's why I'm reluctant to support a random model.

Reply:
OK sounds good.